### ISSUE TITLE
do yo have the same problem of mistaken predicting  class probabilities as lightgbm and xgboost 

### ISSUE THREAD

**Issue by Sandy4321** [catboost/catboost#1275]
Problem:
lightgbm and xgboost  mistakenly predict class probabilities
https://github.com/microsoft/LightGBM/issues/2107

LightGBM people really gave up: they even cheating by closing  issues about this
StrikerRUS commented on Apr 18, 2019
Hmm, autoclosing GitHub feature didn't work again.
Closed in #2113.
***********************************************************
https://xgboost.readthedocs.io/en/latest/tutorials/param_tuning.html
If you care about predicting the right probability
In such a case, you cannot re-balance the dataset 
**************************************************************

------------------------------------------------------------------------
https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html
class_weight (dict, 'balanced' or None, optional (default=None)) – Weights associated with classes in the form {class_label: weight}. Use this parameter only for multi-class classification task; for binary classification task you may use is_unbalance or scale_pos_weight parameters. Note, that the usage of all these parameters will result in poor estimates of the individual class probabilities. You may want to consider performing probability calibration (https://scikit-learn.org/stable/modules/calibration.html) of your model.
---------------------------------------------------------------------------

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
calibration as lightGBM people suggest  is not helping since needs much more data 
https://scikit-learn.org/stable/modules/calibration.html
Probability calibration should be done on new data not used for model fitting.
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

catboost does have
class_weights and scale_pos_weight parameters 
https://catboost.ai/docs/concepts/python-reference_parameters-list.html  

but do yo have the same problem of mistaken predicting  class probabilities

in you documentation this not mentioned at all !

Also can you share algorithm how you calculate class probabilities?

catboost version:latest
Operating System: widows
CPU:

**Comment by Sandy4321** [comment]
also at least for lightGBM  imbalanced data treatment : increase the overall performance metric of your model

as it is written : result in poor estimates of the individual class probabilities
https://lightgbm.readthedocs.io/en/latest/Parameters.html

is_unbalance 🔗︎, default = false, type = bool, aliases: unbalance, unbalanced_sets

used only in binary and multiclassova applications

set this to true if training data are unbalanced

Note: while enabling this should increase the overall performance metric of your model, it will also result in poor estimates of the individual class probabilities
